retry.go

```diff
@@ -0,0 +1,105 @@
+package aviation
+
+import (
+	"context"
+	"time"
+
+	"github.com/jpillora/backoff"
+	"github.com/mongodb/grip"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+func MakeRetryUnaryClientInterceptor(maxRetries int) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		var lastErr error
+		b := &backoff.Backoff{
+			// TODO: figure out the best options for this.
+			Min:    100 * time.Millisecond,
+			Max:    time.Second,
+			Factor: 2,
+		}
+		callCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+	retry:
+		for i := 0; i < maxRetries; i++ {
+			lastErr = invoker(callCtx, method, req, reply, cc, opts...)
+			grip.Infof("GRPC client retry attempt: %d, error: %v", i, lastErr)
+			if lastErr == nil {
+				return nil
+			}
+
+			if !isRetriable(lastErr) {
+				break
+			}
+
+			timer := time.NewTimer(b.Duration())
```

Comment:
defer stop the the timer, as this will leak goroutines. also the defer is probably tricky, because it will run when the function returns not the loop returns

Reply:
put it in the switch statement

Reply:
I think you also want to stop it if it fires... on line 48